internal/pkg/docker/dockercompose/svc.go

```diff
@@ -96,10 +101,68 @@ type exposedPort struct {
 }
 
 // findExposedPort attempts to detect a singular exposed port in the given service and determines if it is publicly exposed.
-func findExposedPort(service *compose.ServiceConfig) (*exposedPort, error) {
-	// TODO: Port handling & exposed port detection, to be implemented in Milestone 3
+func findExposedPort(service *compose.ServiceConfig, workingDir string) (*exposedPort, error) {
+	if len(service.Ports) == 1 {
+		binding := service.Ports[0]
+		port := uint16(binding.Target)
+
+		if binding.Published != "" && strconv.Itoa(int(port)) != binding.Published {
+			return nil, fmt.Errorf("cannot publish the container port %v under a different public port %v in Copilot", port, binding.Published)
+		}
+
+		return &exposedPort{
+			port:   port,
+			public: true,
+		}, nil
+	} else if len(service.Ports) > 1 {
+		return nil, fmt.Errorf("cannot expose more than one public port in Copilot, but %v ports are exposed publicly: %v", len(service.Ports), service.Ports)
+	}
+
+	if len(service.Expose) == 1 {
+		port, err := strconv.Atoi(service.Expose[0])
+		if err != nil {
+			return nil, fmt.Errorf("could not parse exposed port: %w", err)
+		}
+
+		return &exposedPort{
+			port:   uint16(port),
+			public: false,
+		}, nil
+	} else if len(service.Expose) > 1 {
+		return nil, fmt.Errorf("cannot expose more than one port in Copilot, but %v ports are exposed: %v", len(service.Expose), service.Expose)
+	}
+
+	if service.Image != "" || service.Build == nil {
+		// No dockerfile to parse, don't infer any ports.
+		return nil, nil
+	}
+
+	if service.Build.Context == "" {
+		return nil, errors.New("service is missing an image location or Dockerfile path")
+	}
+
+	dockerfilePath := service.Build.Dockerfile
+	if dockerfilePath == "" {
+		dockerfilePath = "Dockerfile"
+	}
+
+	dockerfilePath = filepath.Join(workingDir, service.Build.Context, dockerfilePath)
```

Comment:
This is yet another grey area resulting from not knowing what form/location this tool will take, but does this rely on the user running the conversion in the same place as they'll eventually launch their Copilot services?